modules/src/timestamp.rs

```diff
@@ -50,6 +50,12 @@ pub enum Expiry {
     InvalidTimestamp,
 }
 
+#[cfg(feature = "clock")]
+pub trait Clock {
```

Comment:
Is this trait really needed? All use sites I see in this PR could just use the inherent method on `Time` or `Timestamp`, there is hardly a need to generalize.
Is this meant to be an extension point to provide simulated time in tests?

Reply:
I put this temporarily as I wanted to use the `Time::now()` syntax. Now that tendermint-rs v0.23.5 is released, we can now use `Time::now()` directly without this trait.